From cfc28498e0416fa4579fe919b8fb586985c7b4ea Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 28 Oct 2022 10:13:54 +0300
Subject: [PATCH] [SW-109230] gaudi3: use base_readl to read die id

As part of persistent dccm initializetion code, ROM reads die id
using regmap_readl function which internally makes use of die_id
variable stored in global data structure. In cold reset flow,
read of die_id using regmap_readl will cause uninitialized read
access. This patch add change to make use of base_readl to read
die_id which will aviod access to DCCM before initializing it.

Change-Id: Ib7b3445ee4f3f854759eb483235d82b5477fd78c
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 1e6f548..ed37837 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -461,7 +461,7 @@
  */
 static void arcboot_rom_persistent_dccm_init(void)
 {
-	uint8_t die_id = regmap_readl(mmD0_PARC_LOCAL_CFG_DIE_ID);
+	uint8_t die_id = base_readl(HL_CFG_BASE + mmD0_PARC_LOCAL_CFG_DIE_ID);
 	uint32_t die_cold_rst_flop_2 = HL_CFG_BASE + mmCOLD_RST_FLOPS_2 + die_id * HL_CFG_SPACE_DIE1_OFFSET;
 	struct gaudi3_cold_rst_data cold_rst_data;
 
